### PR TITLE
feat: implement sharing locations

### PR DESCRIPTION
**Pull request by MayerliSantander**
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: apps/client/src/app/pages/map/components/share-location/components/ShareModal.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import CopyLinkButton from './CopyLinkButton';
+
+function ShareModal({ link, setIsOpen, placeName }: ShareModalProps) {
+  const [isSnackbarOpen, setIsSnackbarOpen] = useState(false);
+  const handleClick = () => {
+    navigator.clipboard
+      .writeText(link)
+      .then(() => {
+        setIsSnackbarOpen(true);
+        console.log('Link copied successfully');
```
Comment (isaias-rojas) [inline apps/client/src/app/pages/map/components/share-location/components/ShareModal.tsx:17]: Do not use console.log statements in production code.

File: apps/client/src/app/pages/map/components/share-location/components/ShareLink.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    return <div />;
+  }
+  return (
+    <>
+      {isOpen && (
+        <ShareModal link={link} setIsOpen={setIsOpen} placeName={placeName} />
+      )}
+      <div className='absolute left-6 bottom-40 rounded-full md:left-8 md:bottom-40'>
+        <span>
+          <ShareButton onClick={() => setIsOpen(true)} />
```
Comment (isaias-rojas) [inline apps/client/src/app/pages/map/components/share-location/components/ShareLink.tsx:21]: You should consider opening  the share component when clicking on the same button or elsewhere on the screen

**Review by isaias-rojas — COMMENTED** [review]
Great job @MayerliSantander 
Please consider these suggested changes

**Review by MorveN11 — APPROVED** [review]
Good Job @MayerliSantander @XxMickyxX @CristianQuispia

**Pull request merged**